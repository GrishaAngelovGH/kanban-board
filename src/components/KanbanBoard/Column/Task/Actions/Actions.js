import { Link } from "react-router-dom"

import Button from "react-bootstrap/Button"

import Tooltip from "components/Tooltip"

const bookmarkStyles = {
  "low": "text-warning",
  "medium": "text-success",
  "high": "text-danger"
}

const Actions = ({
  ids: { taskId, columnId, assignedIds, dependencyTasksIds },
  handlers: { handleToggleActiveStatus, handleDelete, handleEdit, handleAssignUser, handleToggleLock },
  statuses: { markedAsDone, isActive, isLocked, isTemplate, hasUsers },
  priority
}) => {
  return (
    <>
      {
        !markedAsDone && assignedIds.length > 0 && !dependencyTasksIds.length && (
          <Button
            size="sm"
            variant="outline-primary"
            disabled={!assignedIds.length}
            className="mb-1"
            onClick={handleToggleActiveStatus}
          >
            {isActive ? "Deactivate" : "Activate"}
          </Button>
        )
      }

      {
        !markedAsDone && (
          <Link
            to={`/task/${columnId}/${taskId}/dependencies`}
            className="btn btn-outline-secondary btn-sm mb-3"
          >
            Manage Dependencies {dependencyTasksIds.length > 0 && `(${dependencyTasksIds.length})`}
          </Link>
        )
      }

      <div className="d-flex justify-content-between">
        <div>
          {!isLocked && <i role="button" onClick={handleDelete} className="bi bi-trash fs-4 text-danger"></i>}
          <i role="button" onClick={handleEdit} className="bi bi-pencil-square fs-4 text-secondary mx-1"></i>
          {!isTemplate && hasUsers && <i role="button" onClick={handleAssignUser} className="bi bi-person-circle fs-4 text-secondary mx-1"></i>}
          <i role="button" onClick={handleToggleLock} className={`bi bi-${isLocked ? "lock-fill" : "unlock-fill"} fs-4 text-secondary mx-1`}></i>
        </div>
        {
          priority.length > 0 && (
            <Tooltip label={`${priority} Priority`} className="text-capitalize">
              <i className={`bi bi-bookmark-fill fs-4 ${bookmarkStyles[priority]}`}></i>
            </Tooltip>
          )
        }
      </div>
    </>
  )
}

export default Actions