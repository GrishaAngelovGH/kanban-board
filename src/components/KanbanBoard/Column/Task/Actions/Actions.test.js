import { render } from "@testing-library/react"

import {
  BrowserRouter as Router,
  Routes,
  Route
} from "react-router-dom"

import Actions from "./Actions"

describe("should render Actions component", () => {
  test("showing the activation button", () => {
    const view = render(
      <Router>
        <Routes>
          <Route
            path="/"
            element={
              <Actions
                ids={{ taskId: "taskId-1", assignedIds: [1, 2], dependencyTasksIds: [] }}
                handlers={{}}
                statuses={{ markedAsDone: false, isActive: false, hasUsers: true }}
                priority={"medium"}
              />
            }
          />
        </Routes>
      </Router>
    )

    expect(view).toMatchSnapshot()
  })

  test("showing the deactivation button", () => {
    const view = render(
      <Router>
        <Routes>
          <Route
            path="/"
            element={
              <Actions
                ids={{ taskId: "taskId-1", columnId: "columnId-1", assignedIds: [1, 2], dependencyTasksIds: [] }}
                handlers={{}}
                statuses={{ markedAsDone: false, isActive: true, hasUsers: true }}
                users={[]}
                priority={"medium"}
              />
            }
          />
        </Routes>
      </Router>
    )

    expect(view).toMatchSnapshot()
  })

  test("not showing the activation/deactivation button", () => {
    const view = render(
      <Router>
        <Routes>
          <Route
            path="/"
            element={
              <Actions
                ids={{ taskId: "taskId-1", columnId: "columnId-1", assignedIds: [], dependencyTasksIds: [] }}
                handlers={{}}
                statuses={{ markedAsDone: true, hasUsers: true }}
                users={[]}
                priority={"medium"}
              />
            }
          />
        </Routes>
      </Router>
    )

    expect(view).toMatchSnapshot()
  })

  test("showing the manage dependencies button", () => {
    const view = render(
      <Router>
        <Routes>
          <Route
            path="/"
            element={
              <Actions
                ids={{ taskId: "taskId-1", columnId: "columnId-1", assignedIds: [1, 2], dependencyTasksIds: [3, 4] }}
                handlers={{}}
                statuses={{ markedAsDone: false, hasUsers: true }}
                users={[]}
                priority={"medium"}
              />
            }
          />
        </Routes>
      </Router>
    )

    expect(view).toMatchSnapshot()
  })

  test("not showing the delete button for locked status", () => {
    const view = render(
      <Router>
        <Routes>
          <Route
            path="/"
            element={
              <Actions
                ids={{ taskId: "taskId-1", columnId: "columnId-1", assignedIds: [1, 2], dependencyTasksIds: [3, 4] }}
                handlers={{}}
                statuses={{ isLocked: true, hasUsers: true }}
                users={[]}
                priority={"medium"}
              />
            }
          />
        </Routes>
      </Router>
    )

    expect(view).toMatchSnapshot()
  })

  test("not showing the user assign button for template", () => {
    const view = render(
      <Router>
        <Routes>
          <Route
            path="/"
            element={
              <Actions
                ids={{ taskId: "taskId-1", columnId: "columnId-1", assignedIds: [1, 2], dependencyTasksIds: [3, 4] }}
                handlers={{}}
                statuses={{ isTemplate: true, hasUsers: false }}
                users={[]}
                priority={"medium"}
              />
            }
          />
        </Routes>
      </Router>
    )

    expect(view).toMatchSnapshot()
  })
})