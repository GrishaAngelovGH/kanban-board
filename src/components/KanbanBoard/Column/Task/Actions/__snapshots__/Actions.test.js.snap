// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Actions component not showing the activation/deactivation button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex justify-content-between"
      >
        <div>
          <i
            class="bi bi-trash fs-4 text-danger"
            role="button"
          />
          <i
            class="bi bi-pencil-square fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-person-circle fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-unlock-fill fs-4 text-secondary mx-1"
            role="button"
          />
        </div>
        <i
          class="bi bi-bookmark-fill fs-4 text-success"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex justify-content-between"
    >
      <div>
        <i
          class="bi bi-trash fs-4 text-danger"
          role="button"
        />
        <i
          class="bi bi-pencil-square fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-person-circle fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-unlock-fill fs-4 text-secondary mx-1"
          role="button"
        />
      </div>
      <i
        class="bi bi-bookmark-fill fs-4 text-success"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Actions component not showing the delete button for locked status 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="btn btn-outline-secondary btn-sm mb-3"
        href="/task/columnId-1/taskId-1/dependencies"
      >
        Manage Dependencies 
        (2)
      </a>
      <div
        class="d-flex justify-content-between"
      >
        <div>
          <i
            class="bi bi-pencil-square fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-person-circle fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-lock-fill fs-4 text-secondary mx-1"
            role="button"
          />
        </div>
        <i
          class="bi bi-bookmark-fill fs-4 text-success"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <a
      class="btn btn-outline-secondary btn-sm mb-3"
      href="/task/columnId-1/taskId-1/dependencies"
    >
      Manage Dependencies 
      (2)
    </a>
    <div
      class="d-flex justify-content-between"
    >
      <div>
        <i
          class="bi bi-pencil-square fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-person-circle fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-lock-fill fs-4 text-secondary mx-1"
          role="button"
        />
      </div>
      <i
        class="bi bi-bookmark-fill fs-4 text-success"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Actions component not showing the user assign button for template 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="btn btn-outline-secondary btn-sm mb-3"
        href="/task/columnId-1/taskId-1/dependencies"
      >
        Manage Dependencies 
        (2)
      </a>
      <div
        class="d-flex justify-content-between"
      >
        <div>
          <i
            class="bi bi-trash fs-4 text-danger"
            role="button"
          />
          <i
            class="bi bi-pencil-square fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-unlock-fill fs-4 text-secondary mx-1"
            role="button"
          />
        </div>
        <i
          class="bi bi-bookmark-fill fs-4 text-success"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <a
      class="btn btn-outline-secondary btn-sm mb-3"
      href="/task/columnId-1/taskId-1/dependencies"
    >
      Manage Dependencies 
      (2)
    </a>
    <div
      class="d-flex justify-content-between"
    >
      <div>
        <i
          class="bi bi-trash fs-4 text-danger"
          role="button"
        />
        <i
          class="bi bi-pencil-square fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-unlock-fill fs-4 text-secondary mx-1"
          role="button"
        />
      </div>
      <i
        class="bi bi-bookmark-fill fs-4 text-success"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Actions component showing the activation button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="mb-1 btn btn-outline-primary btn-sm"
        type="button"
      >
        Activate
      </button>
      <a
        class="btn btn-outline-secondary btn-sm mb-3"
        href="/task/undefined/taskId-1/dependencies"
      >
        Manage Dependencies 
      </a>
      <div
        class="d-flex justify-content-between"
      >
        <div>
          <i
            class="bi bi-trash fs-4 text-danger"
            role="button"
          />
          <i
            class="bi bi-pencil-square fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-person-circle fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-unlock-fill fs-4 text-secondary mx-1"
            role="button"
          />
        </div>
        <i
          class="bi bi-bookmark-fill fs-4 text-success"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      class="mb-1 btn btn-outline-primary btn-sm"
      type="button"
    >
      Activate
    </button>
    <a
      class="btn btn-outline-secondary btn-sm mb-3"
      href="/task/undefined/taskId-1/dependencies"
    >
      Manage Dependencies 
    </a>
    <div
      class="d-flex justify-content-between"
    >
      <div>
        <i
          class="bi bi-trash fs-4 text-danger"
          role="button"
        />
        <i
          class="bi bi-pencil-square fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-person-circle fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-unlock-fill fs-4 text-secondary mx-1"
          role="button"
        />
      </div>
      <i
        class="bi bi-bookmark-fill fs-4 text-success"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Actions component showing the deactivation button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="mb-1 btn btn-outline-primary btn-sm"
        type="button"
      >
        Deactivate
      </button>
      <a
        class="btn btn-outline-secondary btn-sm mb-3"
        href="/task/columnId-1/taskId-1/dependencies"
      >
        Manage Dependencies 
      </a>
      <div
        class="d-flex justify-content-between"
      >
        <div>
          <i
            class="bi bi-trash fs-4 text-danger"
            role="button"
          />
          <i
            class="bi bi-pencil-square fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-person-circle fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-unlock-fill fs-4 text-secondary mx-1"
            role="button"
          />
        </div>
        <i
          class="bi bi-bookmark-fill fs-4 text-success"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      class="mb-1 btn btn-outline-primary btn-sm"
      type="button"
    >
      Deactivate
    </button>
    <a
      class="btn btn-outline-secondary btn-sm mb-3"
      href="/task/columnId-1/taskId-1/dependencies"
    >
      Manage Dependencies 
    </a>
    <div
      class="d-flex justify-content-between"
    >
      <div>
        <i
          class="bi bi-trash fs-4 text-danger"
          role="button"
        />
        <i
          class="bi bi-pencil-square fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-person-circle fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-unlock-fill fs-4 text-secondary mx-1"
          role="button"
        />
      </div>
      <i
        class="bi bi-bookmark-fill fs-4 text-success"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Actions component showing the manage dependencies button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="btn btn-outline-secondary btn-sm mb-3"
        href="/task/columnId-1/taskId-1/dependencies"
      >
        Manage Dependencies 
        (2)
      </a>
      <div
        class="d-flex justify-content-between"
      >
        <div>
          <i
            class="bi bi-trash fs-4 text-danger"
            role="button"
          />
          <i
            class="bi bi-pencil-square fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-person-circle fs-4 text-secondary mx-1"
            role="button"
          />
          <i
            class="bi bi-unlock-fill fs-4 text-secondary mx-1"
            role="button"
          />
        </div>
        <i
          class="bi bi-bookmark-fill fs-4 text-success"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <a
      class="btn btn-outline-secondary btn-sm mb-3"
      href="/task/columnId-1/taskId-1/dependencies"
    >
      Manage Dependencies 
      (2)
    </a>
    <div
      class="d-flex justify-content-between"
    >
      <div>
        <i
          class="bi bi-trash fs-4 text-danger"
          role="button"
        />
        <i
          class="bi bi-pencil-square fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-person-circle fs-4 text-secondary mx-1"
          role="button"
        />
        <i
          class="bi bi-unlock-fill fs-4 text-secondary mx-1"
          role="button"
        />
      </div>
      <i
        class="bi bi-bookmark-fill fs-4 text-success"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
